plugin.yaml

```diff
@@ -759,248 +613,65 @@ node_types:
-              default: false
-            resource_config:
-              description: >
-                Configuration key-value data to be passed as-is to the corresponding
-                Boto3 method. Key names must match the case that Boto3 requires.
-              required: false
-              default: {}
+          inputs: *operation_inputs
         delete:
           implementation: awssdk.cloudify_awssdk.iam.resources.role.delete
-          inputs:
-            aws_resource_id:
-              description: >
-                This overrides the resource_id property (useful for setting the
-                resource ID of a node instance at runtime).
-              type: string
-              required: false
-              default: ~
-            runtime_properties:
-              description: >
-                This overrides any runtime property at runtime. This is a key-value
-                pair / dictionary that will be passed, as-is, to the runtime properties
-                of the running instance.
-              required: false
-              default: ~
-            force_operation:
-              description: >
-                Forces the current operation to be executed regardless
-                if the "use_external_resource" property is set or not.
-              required: false
-              default: false
-            resource_config:
-              description: >
-                Configuration key-value data to be passed as-is to the corresponding
-                Boto3 method. Key names must match the case that Boto3 requires.
-              required: false
-              default: {}
+          inputs: *operation_inputs
 
   # Reference: https://boto3.readthedocs.io/en/latest/reference/services/iam.html#IAM.Client.put_role_policy
   cloudify.nodes.aws.iam.RolePolicy:
     derived_from: cloudify.nodes.Root
     properties:
-      use_external_resource:
-        description: *use_external_resource_desc
-        type: boolean
-        default: false
+      <<: *external_resource
+      <<: *client_config
+      <<: *resource_id
+      resource_config:
+        description: >
+          Configuration key-value data to be passed as-is to the corresponding
+          Boto3 method. Key names must match the case that Boto3 requires.
+        type: cloudify.datatypes.aws.iam.RolePolicy.config
+        required: false
+    interfaces:
+      cloudify.interfaces.lifecycle:
+        create:
+          implementation: awssdk.cloudify_awssdk.iam.resources.role_policy.create
+          inputs: *operation_inputs
+        delete:
+          implementation: awssdk.cloudify_awssdk.iam.resources.role_policy.delete
+          inputs: *operation_inputs
+
+  # Reference: http://boto3.readthedocs.io/en/latest/reference/services/iam.html#IAM.Client.create_instance_profile
+  cloudify.nodes.aws.iam.InstanceProfile:
+    derived_from: cloudify.nodes.Root
+    properties:
+      <<: *external_resource
+      <<: *client_config
```

Comment:
@EarthmanT I think you miss here the ``resource_id``, you need to update it and use ``<<:*resource_id``